src/main/java/beam/physsim/jdeqsim/AgentSimToPhysSimPlanConverter.java

```diff
@@ -116,6 +116,23 @@ private void preparePhysSimForNewIteration() {
         jdeqsimPopulation = PopulationUtils.createPopulation(agentSimScenario.getConfig());
     }
 
+    private void validateForInvalidNodeChains() {
+        List<Link> errorLinks = new ArrayList<>();
+        Map<Id<Link>, ? extends Link> links = agentSimScenario.getNetwork().getLinks();
+        agentSimScenario.getPopulation().getPersons().values().
+                forEach(f -> f.getSelectedPlan().getPlanElements().forEach(pe -> {
+                    if(pe instanceof Leg) {
+                        Leg leg = (Leg) pe;
+                        Link start = links.get(leg.getRoute().getStartLinkId());
```

Comment:
From the description of #1731 it seems that this should be comparing two separate links, not the start and end of the same one. More like the end of the first, and the start of the next. Also, if there is any way to perform this check other than a double `for` that would be preferred for performance reasons.